packages/plugin-ext/src/main/browser/view/tree-view-widget.tsx

```diff
@@ -549,38 +550,42 @@ export class TreeViewWidget extends TreeViewWelcomeWidget {
             const menu = this.menus.getMenu(VIEW_ITEM_INLINE_MENU);
-            const arg = this.toTreeViewSelection(node);
+            const args = this.toContextMenuArgs(node);
             const inlineCommands = menu.children.filter((item): item is ActionMenuNode => item instanceof ActionMenuNode);
             const tailDecorations = super.renderTailDecorations(node, props);
             return <React.Fragment>
                 {inlineCommands.length > 0 && <div className={TREE_NODE_SEGMENT_CLASS + ' flex'}>
-                    {inlineCommands.map((item, index) => this.renderInlineCommand(item, index, this.focusService.hasFocus(node), arg))}
+                    {inlineCommands.map((item, index) => this.renderInlineCommand(item, index, this.focusService.hasFocus(node), args))}
                 </div>}
                 {tailDecorations !== undefined && <div className={TREE_NODE_SEGMENT_CLASS + ' flex'}>{tailDecorations}</div>}
             </React.Fragment>;
         });
     }
 
-    toTreeViewSelection(node: TreeNode): TreeViewSelection {
-        return { treeViewId: this.id, treeItemId: node.id };
+    toTreeViewItemReference(node: TreeNode): TreeViewItemReference {
+        return { viewId: this.id, itemId: node.id };
     }
 
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    protected renderInlineCommand(node: ActionMenuNode, index: number, tabbable: boolean, arg: any): React.ReactNode {
+    protected renderInlineCommand(node: ActionMenuNode, index: number, tabbable: boolean, args: any[]): React.ReactNode {
         const { icon } = node;
-        if (!icon || !this.commands.isVisible(node.command, arg) || !node.when || !this.contextKeys.match(node.when)) {
+        if (!icon || !this.commands.isVisible(node.command, ...args) || !node.when || !this.contextKeys.match(node.when)) {
             return false;
         }
         const className = [TREE_NODE_SEGMENT_CLASS, TREE_NODE_TAIL_CLASS, icon, ACTION_ITEM, 'theia-tree-view-inline-action'].join(' ');
         const tabIndex = tabbable ? 0 : undefined;
         return <div key={index} className={className} title={node.label} tabIndex={tabIndex} onClick={e => {
             e.stopPropagation();
-            this.commands.executeCommand(node.command, arg);
+            this.commands.executeCommand(node.command, ...args);
         }} />;
     }
 
-    protected override toContextMenuArgs(node: SelectableTreeNode): [TreeViewSelection] {
-        return [this.toTreeViewSelection(node)];
+    protected override toContextMenuArgs(target: SelectableTreeNode): [TreeViewItemReference, TreeViewItemReference[]] | [TreeViewItemReference] {
+        if (this.options.multiSelect) {
+            return [this.toTreeViewItemReference(target), this.model.selectedNodes.map(node => this.toTreeViewItemReference(node))];
+        } else {
+            return [this.toTreeViewItemReference(target)];
+        }
     }
```

Comment:
I was unable to understand which handler would consume the returned values... Ultimately they seem to be forwarded to `this.commands.executeCommand(node.command, ...args);` line 579 of the same file but that doesn't help me either.

It doesn't seem like your sample VS Code Extension contributes any inline command for us to test this code, right?

Reply:
Command arguments are described [here](https://code.visualstudio.com/api/references/vscode-api#TreeViewOptions&lt;T&gt;). The "Print Selection" from the extension command spews out the command arguments to the console.

Reply:
I don't see the documentation about said arguments in the link you shared...

I finally managed to find the equivalent code in VS Code that may explain what context menu items expect to invoke commands with: https://github.com/microsoft/vscode/blob/d5379b5e34e4bb0f5ddb33688da91f787ebc159f/src/vs/workbench/browser/parts/views/treeView.ts#L1192-L1210

Reply:
TreeViewOptions.canSelectMany doc says: 

>Whether the tree supports multi-select. When the tree supports multi-select and a command is executed from the tree, the first argument to the command is the tree item that the command was executed on and the second argument is an array containing all selected tree items.

Reply:
I'm not sure what you mean by "inline commands".